nameserver/dns.go

```diff
@@ -70,14 +70,14 @@ func (d *DNSServer) listen(address string) error {
 	if err != nil {
 		return err
 	}
-	udpServer := &dns.Server{PacketConn: udpListener, Handler: d.createMux(d.udpClient, minUDPSize)}
+	udpServer := &dns.Server{PacketConn: udpListener, Handler: d.createMux(d.udpClient, minUDPSize).mux}
```

Comment:
I changed the mux to an embedded struct, so the handler is now a ServeMux.  What do you think?